src/monitors/WhammMonitor.v3

```diff
@@ -347,6 +354,56 @@ class WhammMonitor(named_modules: Array<NamedModule>) extends Monitor {
 		}
 		return args;
 	}
+	private def getNumTargets(op: Opcode, cv: CodeValidator, expType: ValueType) -> WhammArg {
+		var arg: WhammArg;
+		match (op) {
+			BR_TABLE => {
+				def parser = dupParser(cv);
+				// num_targets is the number of branches (not including the default branch)
+				for (imm in op.imms) {
+					match (imm) {
+						LABELS            => {
+							def labels = parser.readLabels();
+							arg = WhammArg.Val(Values.box_i(labels.length - 1));
+							if (expType != ValueType.I32) {
+								err.at(cv.parser.decoder).WhammProbeError("whamm probe", "type mismatch on num_targets argument");
+							}
+						}
+						_ => ; // ignore
+					}
+				}
+			}
+			_ => {
+				err.at(cv.parser.decoder).WhammProbeError("whamm probe", "num_targets variable only in-scope for br_table opcode.");
+			}
+		}
+		return arg;
+	}
+	private def getOffset(op: Opcode, cv: CodeValidator, expType: ValueType) -> WhammArg {
+		def imms = op.imms;
+		def parser = dupParser(cv);
+		var arg: WhammArg;
+
+		var set = false;
+		for (idx < imms.length) {
```

Comment:
This is a little rickety; it looks like it won't work if the MEMARG is not the first immediate. There's also a growing amount of Whamm-specific bytecode decoding going on that could probably benefit from factoring into a more common place.